### PR TITLE
Add validation for the search backpressure cancellation settings

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description
Updating the cluster level setting `search_backpressure.cancellation_burst`(deprecated), `search_backpressure.search_task.cancellation_burst` or `search_backpressure.search_shard_task.cancellation_burst` to an non-default value cause the cluster crash, the reason is that the value of `cancellationRate` is always 0 here:
https://github.com/opensearch-project/OpenSearch/blob/2b843051a4e71d79fd8fb94014cc8a41a0f97560/server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureState.java#L101
, so an exception is thrown here:
https://github.com/opensearch-project/OpenSearch/blob/2b843051a4e71d79fd8fb94014cc8a41a0f97560/server/src/main/java/org/opensearch/common/util/TokenBucket.java#L51.

Just like `cancellationRatio`, we also need to get the value of `cancellationRate` from the cluster settings and pass it to the constructor method of `SearchBackpressureState`.

Another issue is that we don't have some validations for the setting `cancellationRate` and `cancellationRatio`, so when updating these two settings to 0, the same exception "rate must be greater than zero" is also thrown and then crash the cluster.

### Related Issues

https://github.com/opensearch-project/OpenSearch/issues/15495

### Check List
- [x] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaobinlong** [comment]
@reta @dblock could you help to review this PR, thank you!

File: server/src/main/java/org/opensearch/search/backpressure/settings/SearchBackpressureSettings.java [hunk 1]
```diff
@@ -63,6 +63,11 @@ private static class Defaults {
         Defaults.CANCELLATION_RATIO,
         0.0,
         1.0,
+        value -> {
```
Comment (reta) [inline server/src/main/java/org/opensearch/search/backpressure/settings/SearchBackpressureSettings.java:66]: This validation looks redundant, the `Setting.doubleSetting()` has min and max bounds that are set to `0.0`and `1.0`  respectively, didn't validation trigger?

Comment (gaobinlong) [reply server/src/main/java/org/opensearch/search/backpressure/settings/SearchBackpressureSettings.java:66]: Yes, the problem is that 0.0 is not acceptable, users can set any value greater than 0.0, but no minimal bound exists.

Comment (reta) [reply server/src/main/java/org/opensearch/search/backpressure/settings/SearchBackpressureSettings.java:66]: In this case we have three choices:
 - make sure 0.0 is handled properly down the line
 - have reasonable minimal value (fe. 0.01)
 - use different factory method that does use custom validation without min/max bounds

Comment (gaobinlong) [reply server/src/main/java/org/opensearch/search/backpressure/settings/SearchBackpressureSettings.java:64]: Choice 3 makes more sense, please take a look at the new change, thanks!

File: server/src/main/java/org/opensearch/search/backpressure/settings/SearchShardTaskSettings.java [hunk 2]
```diff
@@ -48,8 +48,14 @@ private static class Defaults {
...
         "search_backpressure.search_shard_task.cancellation_ratio",
         SearchBackpressureSettings.SETTING_CANCELLATION_RATIO,
-        0.0,
-        1.0,
+        value -> {
+            if (value <= 0.0) {
+                throw new IllegalArgumentException("search_backpressure.search_shard_task.cancellation_ratio must be > 0");
+            }
+            if (value > 1.0) {
+                throw new IllegalArgumentException("search_backpressure.search_shard_task.cancellation_ratio must <= 1.0");
```
Comment (reta) [inline server/src/main/java/org/opensearch/search/backpressure/settings/SearchShardTaskSettings.java:56]:
```suggestion
                throw new IllegalArgumentException("search_backpressure.search_shard_task.cancellation_ratio must be <= 1.0");
```

**Review by reta — APPROVED** [review]
LGTM, thanks @gaobinlong ! @andrross any concerns you might have? thanks!

File: rest-api-spec/src/main/resources/rest-api-spec/test/cluster.put_settings/10_basic.yml [hunk 3]
```diff
@@ -98,3 +98,116 @@
   - match: {error.root_cause.0.type: "illegal_argument_exception"}
   - match: { error.root_cause.0.reason: "Invalid SearchBackpressureMode: monitor-only" }
   - match: { status: 400 }
+
+
+---
+"Test setting search backpressure cancellation settings":
+  - skip:
+      version: "- 2.99.99"
+      reason: "Fixed in 3.0.0"
```
Comment (sandeshkr419) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cluster.put_settings/10_basic.yml:107]: I see backport label added, should this be skipped until 2.17.99 only?

Comment (reta) [reply rest-api-spec/src/main/resources/rest-api-spec/test/cluster.put_settings/10_basic.yml:107]: After backport would be safer to change it

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
Sorry @gaobinlong , completely forgot about this change

**Comment by reta** [comment]
@gaobinlong could you please backport to `2.x` manually? thank you!

**Review by UglyJoevIne — APPROVED** [review]

**Review by UglyJoevIne — APPROVED** [review]

**Pull request merged**